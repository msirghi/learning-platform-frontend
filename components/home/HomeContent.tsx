import React from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import { NewsCard } from './NewsCard';
import { newsMock } from '../../mocks/newsMocks';
import { ScheduleCard } from './ScheduleCard';
import { useWindowSize } from '../common/hooks/useWindowResize';
import styles from '../../styles/modules/Home.module.scss';
import { useTranslation } from '../../i18n';
import { NoContent } from './NoContent';

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      display: 'flex',
      justifyContent: 'space-around'
    },
    grid: {
      maxWidth: 1400
    }
  })
);

/**
 * Home content wrapper.
 *
 * @version 0.1
 * @author [Sirghi Mihail](https://github.com/msirghi)
 */
export const HomeContent: React.FC = () => {
  const classes = useStyles();
  const [width] = useWindowSize();
  const { t } = useTranslation();

  return (
    <div className={classes.root}>
      <Grid container spacing={width < 900 ? 5 : 10} className={classes.grid}>
        <Grid item xs={12} sm={12} md={12} lg={6}>
          <ScheduleCard />
        </Grid>
        <Grid item xs={12} sm={12} md={12} lg={6}>
          <NewsCard title={t('home:groupNewsTitle')} news={newsMock} />
          <div className={styles.cardNewsMargin}>
            <NewsCard title={t('home:unversityNewsTitle')} news={newsMock} />
          </div>
        </Grid>
      </Grid>
    </div>
  );
};
