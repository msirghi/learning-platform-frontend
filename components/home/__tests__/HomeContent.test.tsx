import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { HomeContent } from '../HomeContent';
import Grid from '@material-ui/core/Grid';
import MockDate from 'mockdate';

describe('HomeContent component', () => {
  const initialWidth = global.innerWidth;

  beforeAll(() => {
    MockDate.set('2000-11-22');
  });

  afterAll(() => {
    global.innerWidth = initialWidth;
    MockDate.reset();
    global.dispatchEvent(new Event('resize'));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<HomeContent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a Grid with spacing of 10 if the screen width is greater than 768px', () => {
    global.innerWidth = 1200;
    global.dispatchEvent(new Event('resize'));
    const wrapper = mount(<HomeContent />);
    const grid = wrapper.find(Grid).at(0);
    expect(grid.props().spacing).toBe(10);
  });

  it('should have a Grid with spacing of 5 if the screen width is less than 768px', () => {
    global.innerWidth = 700;
    global.dispatchEvent(new Event('resize'));
    const wrapper = mount(<HomeContent />);
    const grid = wrapper.find(Grid).at(0);
    expect(grid.props().spacing).toBe(5);
  });
});
