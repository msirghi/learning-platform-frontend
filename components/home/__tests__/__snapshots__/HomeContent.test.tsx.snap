// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContent component should match the snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-grid-2"
    container={true}
    spacing={5}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={6}
      md={12}
      sm={12}
      xs={12}
    >
      <ScheduleCard />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={6}
      md={12}
      sm={12}
      xs={12}
    >
      <NewsCard
        news={
          Array [
            Object {
              "date": 2000-11-22T00:00:00.000Z,
              "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debitis in.",
              "id": "1",
              "title": "News 1",
            },
            Object {
              "date": 2000-11-22T00:00:00.000Z,
              "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Officiis numquam nobis, nihil quis error repudiandae.",
              "id": "2",
              "title": "News 2",
            },
          ]
        }
        title="home:groupNewsTitle"
      />
      <div
        className="cardNewsMargin"
      >
        <NewsCard
          news={
            Array [
              Object {
                "date": 2000-11-22T00:00:00.000Z,
                "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Dignissimos nihil, commodi alias at debitis in.",
                "id": "1",
                "title": "News 1",
              },
              Object {
                "date": 2000-11-22T00:00:00.000Z,
                "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Officiis numquam nobis, nihil quis error repudiandae.",
                "id": "2",
                "title": "News 2",
              },
            ]
          }
          title="home:unversityNewsTitle"
        />
      </div>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
